paddle/fluid/operators/optimizers/adam_op.h

```diff
@@ -51,9 +51,47 @@ class AdamOp : public framework::OperatorWithKernel {
 struct GPUAdam;
 struct CPUAdam;
 
+template <typename T>
+struct BetaPowFunctor;
+
 template <typename T, typename Flavour>
 struct AdamFunctor;
 
+template <typename T>
+struct BetaPowFunctor {
```

Comment:
Use "class" instead of "struct" for BetaPowFunctor, same for AdamFunctor

Use a struct only for passive objects that carry data; everything else is a class. You are implementing Functor here.

Reference:
https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes

Reply:
Thanks, fixed.